### DISCUSSION TITLE
How do we properly paste in paragraphs of text?

### DISCUSSION THREAD

**Discussion by abea** [GrapesJS/grapesjs#5681] [Q&A] [answered]
If someone wrote several paragraphs in a Google doc or MS Word, that text will be formatted in paragraph tags. It seems the GrapesJS default for a text block is an empty div (I'm using the blocks basics plugin). Pasting that text into a text block will add sub-blocks which have their own hover UI nested inside.

Then if I edit the test and hit Enter for a new paragraph, it'll add break tags instead of a new paragraph. This all makes for really messy markup and seemingly not great a11y for screen readers.

Is there a best practice for this I haven't come across yet?

**Comment by mosh-tudor** [comment, marked as answer]
@abea 

For more advanced text editing, you need to integrate a WYSIWYG editor (Froala/CKEditor/..).

**Reply by artf** [reply]
Thanks @mosh-tudor I really appreciate the help you're providing to the community 🙇‍♂️
